pkg/target/manager.go

```diff
@@ -282,8 +292,122 @@ func (m *managerImpl) validateSeed(ctx context.Context, seed *gardencorev1beta1.
 	return nil
 }
 
+func domainsFromConfiguration(m *managerImpl) (map[string]string, map[string]string, error) {
+	domainRegexp, err := regexp.Compile(`(http://|https://)?(.+\..+)`)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to compile regular expression: %v", err)
+	}
+
+	dashboardDomains := make(map[string]string)
+	shootDomains := make(map[string]string)
+
+	for _, g := range m.Configuration().Gardens {
+		match := domainRegexp.FindStringSubmatch(g.DashboardDomain)
+		if match != nil {
+			dashboardDomains[g.Name] = match[2]
+		}
+
+		match = domainRegexp.FindStringSubmatch(g.ShootDomain)
+		if match != nil {
+			shootDomains[g.Name] = match[2]
+		}
+	}
+
+	return dashboardDomains, shootDomains, nil
+}
+
+func matchGardenDomain(domains map[string]string, domain string) (string, string) {
+	for garden, gDomain := range domains {
+		if strings.Contains(domain, gDomain) {
+			return garden, domain
+		}
+	}
+
+	return "", ""
+}
+
+func targetFlagsFromDomain(ctx context.Context, m *managerImpl, t *targetImpl, name string) (TargetFlags, error) {
+	dashboardDomains, shootDomains, err := domainsFromConfiguration(m)
+	if err != nil {
+		return nil, fmt.Errorf("invalid domain format in configured gardens: %w", err)
+	}
+
+	if len(dashboardDomains) == 0 && len(shootDomains) == 0 {
+		return nil, fmt.Errorf("no domain found in configured gardens. Shoot targeting via domain requires shoot and / or dashboard domain configuration for each garden that you want to enable domain targeting for")
```

Comment:
if you need to format your error message use `fmt.Errorf()`, if you don't use `errors.New()`.
-> in this case, use `errors.New()`